### DISCUSSION TITLE
Microsoft Security Advisory CVE-2023-36038: .NET Denial of Service Vulnerability

### DISCUSSION THREAD

**Discussion by rbhanda** [dotnet/aspnetcore#64483] [Security advisories]
# Microsoft Security Advisory CVE-2023-36038: .NET Denial of Service Vulnerability

## <a name="executive-summary"></a>Executive summary

Microsoft is releasing this security advisory to provide information about a vulnerability in ASP.NET Core 8.0 RC2. This advisory also provides guidance on what developers can do to update their applications to address this vulnerability.

A vulnerability exists in ASP.NET IIS where a remote  unauthenticated user can issue specially crafted requests to a .NET application which may result in denial of service.

## Announcement

Announcement for this issue can be found at  https://github.com/dotnet/announcements/issues/286

### <a name="mitigation-factors"></a>Mitigation factors

This vulnerability only affects ASP.NET Core applications hosted in IIS. ASP.NET Core applications hosted by Kestrel are not affected.

## <a name="affected-software"></a>Affected software

* Any ASP.NET Core 8.0 application running on .NET 8.0 RC2 or earlier.

### <a name="how-affected"></a>How do I know if I am affected?

If you have a runtime or SDK with a version listed, or an affected package listed in [affected software](#affected-software), you're exposed to the vulnerability.

### <a name="how-fix"></a>How do I fix the issue?

* To fix the issue please install the latest version of  .NET 8.0. If you have installed one or more .NET SDKs through Visual Studio, Visual Studio will prompt you to update Visual Studio, which will also update your .NET  SDKs.
* If you are using one of the affected packages, please update to the patched version listed above.
* If you have .NET 6.0 or greater installed, you can list the versions you have installed by running the `dotnet --info` command. You will see output like the following;

```
.NET Core SDK (reflecting any global.json):

 Version:   8.0.100
 Commit:    8473146e7d

Runtime Environment:

 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.100\

Host (useful for support):

  Version: 8.0.0
  Commit:  8473146e7d

.NET Core SDKs installed:

  8.0.100 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:

  Microsoft.AspNetCore.App 8.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

* If you're using .NET 8.0, you should download and install .NET 8.0.0  Runtime  or .NET 8.0.100 SDK (for Visual Studio 2022 v17.8) from https://dotnet.microsoft.com/download/dotnet-core/8.0.

If you've deployed [self-contained applications](https://docs.microsoft.com/dotnet/core/deploying/#self-contained-deployments-scd) targeting any of the impacted versions, these applications are also vulnerable and must be recompiled and redeployed.

### Reporting Security Issues

If you have found a potential security issue in .NET 8.0 or .NET 7.0 or .NET 6.0, please email details to secure@microsoft.com. Reports may qualify for the Microsoft .NET Core & .NET 5 Bounty. Details of the Microsoft .NET Bounty Program including terms and conditions are at <https://aka.ms/corebounty>.

### Support

You can ask questions about this issue on GitHub in the .NET GitHub organization. The main repos are located at https://github.com/dotnet/runtime and https://github.com/dotnet/aspnet/. The Announcements repo (https://github.com/dotnet/Announcements) will contain this bulletin as an issue and will include a link to a discussion issue. You can ask questions in the linked discussion issue.

### Disclaimer

The information provided in this advisory is provided "as is" without warranty of any kind. Microsoft disclaims all warranties, either express or implied, including the warranties of merchantability and fitness for a particular purpose. In no event shall Microsoft Corporation or its suppliers be liable for any damages whatsoever including direct, indirect, incidental, consequential, loss of business profits or special damages, even if Microsoft Corporation or its suppliers have been advised of the possibility of such damages. Some states do not allow the exclusion or limitation of liability for consequential or incidental damages so the foregoing limitation may not apply.

### External Links

[CVE-2023-36038]( https://www.cve.org/CVERecord?id=CVE-2023-36038)

### Revisions

V1.0 (November 14, 2023): Advisory published.

_Version 1.0_

_Last Updated 2023-11-14_

**Comment by veintrop** [comment]
Hello,

Can you please clarify what products are affected.
I see that "A vulnerability exists in **ASP.NET IIS** where a remote unauthenticated user can issue specially crafted requests to a .NET application which may result in denial of service".
But **.net** update is listed as a possible remediation: "To fix the issue please install the latest version of **.NET 8.0**".
Microsoft update guide also lists both .net and ASP.net as possible solutions: https://msrc.microsoft.com/update-guide/en-US/vulnerability/CVE-2023-36038

Is it a shared SDK problem?

What if I have only .net installed with version that is lower than 8.0. Am I still affected? Is there any routine that will affect my applications the same way? Or I have to have (and use) ASP net framework?
I understand that it is probably safer to update anyway, but I would be grateful for a detailed explanation.
